src/main/java/uk/ac/ebi/embl/api/validation/submission/SubmissionValidationPlan.java

```diff
@@ -151,149 +160,185 @@ public void execute() throws ValidationEngineException {
 			check = new ChromosomeListFileValidationCheck(options);
-			for(SubmissionFile chromosomeListFile:options.submissionFiles.get().getFiles(FileType.CHROMOSOME_LIST))
-			{
+			for (SubmissionFile chromosomeListFile : options.submissionFiles.get().getFiles(FileType.CHROMOSOME_LIST)) {
 				fileName= chromosomeListFile.getFile().getName();
-
-				if(!check.check(chromosomeListFile))
-					throwValidationCheckException(FileType.CHROMOSOME_LIST,chromosomeListFile);
+				result = check.check(chromosomeListFile);
+				if (!result.isValid()) {
+					result.setDefaultOrigin(new DefaultOrigin(getValidationErrorMessage(FileType.CHROMOSOME_LIST, chromosomeListFile, check)));
+					return result;
+				}
 			}
-		}catch(Exception e)
-		{
-			throwValidationEngineException(FileType.CHROMOSOME_LIST,e,fileName);
+		} catch (ValidationEngineException e) {
+			throwValidationEngineException(FileType.CHROMOSOME_LIST, e, fileName);
 		}
+		return result;
+	}
+	private String getValidationErrorMessage(FileType fileTpe, SubmissionFile submissionFile, FileValidationCheck fileCheck)  {
+		return String.format("%s file validation failed : %s, Please see the error report: %s",
+				fileTpe.name().toLowerCase(), submissionFile.getFile().getName() , fileCheck.getReportFile(submissionFile).toFile());
 	}
 
-	private void validateFasta() throws ValidationEngineException
+	private ValidationResult validateFasta() throws ValidationEngineException
 	{
-		if(options.processDir.isPresent()&&Files.exists(Paths.get(String.format("%s%s%s",options.processDir.get(),File.separator,fastaFlagFileName))))
-			return;
+		ValidationResult result = new ValidationResult();
 
+		if(options.processDir.isPresent()&&Files.exists(Paths.get(String.format("%s%s%s",options.processDir.get(),File.separator,fastaFlagFileName))))
+			return result;
 		String fileName=null;
-
-		try
-		{
+		try {
 			check = new FastaFileValidationCheck(options);
-			for(SubmissionFile fastaFile:options.submissionFiles.get().getFiles(FileType.FASTA))
-			{
+			for (SubmissionFile fastaFile : options.submissionFiles.get().getFiles(FileType.FASTA)) {
 				fileName= fastaFile.getFile().getName();
-				if(sequenceDB!=null)
+				if (sequenceDB != null)
 					check.setSequenceDB(sequenceDB);
-				if(contigDB!=null)
+				if (contigDB != null)
 					check.setContigDB(contigDB);
-				if(!check.check(fastaFile))
-					throwValidationCheckException(FileType.FASTA,fastaFile);
-				else if(!options.isRemote)
-				     flagValidation(FileType.FASTA);
-
+				result = check.check(fastaFile);
+				if (!result.isValid()) {
+					result.setDefaultOrigin(new DefaultOrigin(getValidationErrorMessage(FileType.FASTA, fastaFile, check)));
+					return result;
+				}
 			}
-		}catch(Exception e)
-		{
+			if (!options.isRemote)
+				flagValidation(FileType.FASTA);
+		} catch(ValidationEngineException e) {
 			throwValidationEngineException(FileType.FASTA,e,fileName);
+		}catch(IOException e)
+		{
+			throw new ValidationEngineException(e);
 		}
```

Comment:
Now sure why we don't just do:

catch(Exception e)
{
throwValidationEngineException(FileType. FASTA,e,"master.dat");
}